pkg/policy/manager.go

```diff
@@ -73,7 +73,12 @@ func (m *ManagerImpl) Run(ctx context.Context, request *ExecuteRequest) (*Execut
 		totalQueriesToRun = request.Policy.TotalQueries()
 		finishedQueries   = 0
 	)
-
+	filteredPolicy := request.Policy.Filter(request.Policy.meta.subPolicy)
+	if !filteredPolicy.HasChecks() {
+		m.logger.Error("policy/query not found with provided sub-policy selector", "selector", request.Policy.meta.subPolicy, "available_policies", filteredPolicy.Policies.All())
+		return nil, fmt.Errorf("%s//%s: %w", request.Policy.Name, request.Policy.meta.subPolicy, ErrPolicyOrQueryNotFound)
+	}
+	totalQueriesToRun = filteredPolicy.TotalQueries()
```

Comment:
This is rearranged so that the total Queries to run is calculated after filtering and not before